src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/versionMatcher/versionObjectDpkg.hpp

```diff
@@ -0,0 +1,318 @@
+     * @param dpkgStruct The parsed version.
+     *
+     * @retval  true  On success.
+     * @retval  false On failure.
+     */
+    static bool parseversion(const char* string, struct DpkgVersion* dpkgStruct)
+    {
+        const char *end, *ptr;
+
+        /* Trim leading and trailing space. */
+        while (*string && std::isspace(*string)) string++;
+
+        if (!*string)
+            return false;
+
+        /* String now points to the first non-whitespace char. */
+        end = string;
+        /* Find either the end of the string, or a whitespace char. */
+        while (*end && !std::isspace(*end)) end++;
+        /* Check for extra chars after trailing space. */
+        ptr = end;
+        while (*ptr && std::isspace(*ptr)) ptr++;
+        if (*ptr)
+            return false;
+
+        auto colon = std::strchr(string, ':');
+        if (colon != nullptr)
+        {
+            long epoch;
+            char* eepochcolon;
+
+            errno = 0;
+            epoch = std::strtol(string, &eepochcolon, 10);
+            if (string == eepochcolon)
+                return false;
+            if (colon != eepochcolon)
+                return false;
+            if (epoch < 0)
+                return false;
+            if (epoch > INT_MAX || errno == ERANGE)
+                return false;
+            if (!*++colon)
+                return false;
+            string = colon;
+            dpkgStruct->epoch = epoch;
+        }
+        else
+        {
+            dpkgStruct->epoch = 0;
+        }
+        dpkgStruct->version = std::string(string, end - string);
+        auto hyphen = const_cast<char*>(std::strchr(dpkgStruct->version.c_str(), '-'));
+        if (hyphen != nullptr)
+        {
+            *hyphen++ = '\0';
+            if (*hyphen == '\0')
+                return false;
+        }
+        dpkgStruct->revision = hyphen ? std::string(hyphen) : "";
+
+        /* XXX: Would be faster to use something like cisversion and cisrevision. */
+        ptr = dpkgStruct->version.c_str();
+        if (!*ptr)
+            return false;
+        if (!std::isdigit(*ptr++))
+            return false;
+        for (; *ptr; ptr++)
+        {
+            if (!std::isdigit(*ptr) && !std::isalpha(*ptr) && strchr(".-+~:", *ptr) == NULL)
+                return false;
+        }
+        for (ptr = dpkgStruct->revision.c_str(); *ptr; ptr++)
+        {
+            if (!std::isdigit(*ptr) && !std::isalpha(*ptr) && strchr(".+~", *ptr) == NULL)
+                return false;
+        }
+
+        return true;
+    }
+
+public:
+    /**
+     * @brief Static method to match a version string to a DpkgVersion object.
+     *
+     * @param version version string to match.
+     * @param output DpkgVersion object to store the result.
+     * @return true/false according to match condition.
+     */
+    static bool match(const std::string& version, DpkgVersion& output)
+    {
+        return parseversion(version.c_str(), &output);
```

Comment:
Why don't you put the content of the method "parseversion" directly here?

Reply:
Moved!